### PR TITLE
[SMALLFIX] Use a heartbeat thread to update metrics.

### PR DESCRIPTION
**Pull request by calvinjia**
Calling `MultiValueMetricsUpdate` incurs some costs, such as querying ufs for available space. Also, this gets call on each heartbeat which can be very frequent. Instead, do this on a best effort basis periodically.

### CODE REVIEW

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 1]
```diff
@@ -958,6 +958,13 @@ public String toString() {
           .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
           .setScope(Scope.MASTER)
           .build();
+  public static final PropertyKey MASTER_CLUSTER_METRICS_UPDATE_INTERVAL =
+      new Builder(Name.MASTER_CLUSTER_METRICS_UPDATE_INTERVAL)
+          .setDefaultValue("1m")
+          .setDescription("The interval for periodically updating the cluster level metrics.")
+          .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
+          .setScope(Scope.MASTER)
```
Comment (aaudiber) [inline core/common/src/main/java/alluxio/PropertyKey.java]: perhaps make this hidden? Not sure whether users would ever want to change it.

File: core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java [hunk 2]
```diff
@@ -53,6 +57,22 @@
   private final Set<MultiValueMetricsAggregator> mMultiValueMetricsAggregatorRegistry =
       new HashSet<>();
   private final MetricsStore mMetricsStore;
+  private final HeartbeatThread mClusterMetricsUpdater;
+
+  /**
+   * Heartbeat executor that updates the cluster metrics.
+   */
+  private class ClusterMetricsUpdater implements HeartbeatExecutor {
```
Comment (aaudiber) [inline core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java]: I think we usually put inner classes towards the bottom of the file

File: core/common/src/main/java/alluxio/PropertyKey.java (hunk 1 above)
Comment (calvinjia) [reply core/common/src/main/java/alluxio/PropertyKey.java]: Updated, thanks.

File: core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java (hunk 2 above)
Comment (calvinjia) [reply core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java]: Updated, thanks.

**Pull request merged**